clients/src/main/java/org/apache/kafka/common/requests/RequestContext.java

```diff
@@ -38,19 +39,37 @@
     public final KafkaPrincipal principal;
     public final ListenerName listenerName;
     public final SecurityProtocol securityProtocol;
-
-    public RequestContext(RequestHeader header,
-                          String connectionId,
-                          InetAddress clientAddress,
-                          KafkaPrincipal principal,
-                          ListenerName listenerName,
-                          SecurityProtocol securityProtocol) {
+    public final String clientSoftwareName;
+    public final String clientSoftwareVersion;
+
+    public RequestContext(
+            RequestHeader header,
+            String connectionId,
+            InetAddress clientAddress,
+            KafkaPrincipal principal,
+            ListenerName listenerName,
+            SecurityProtocol securityProtocol,
+            String clientSoftwareName,
+            String clientSoftwareVersion) {
         this.header = header;
         this.connectionId = connectionId;
         this.clientAddress = clientAddress;
         this.principal = principal;
         this.listenerName = listenerName;
         this.securityProtocol = securityProtocol;
+        this.clientSoftwareName = clientSoftwareName;
+        this.clientSoftwareVersion = clientSoftwareVersion;
+    }
+
+    public RequestContext(
```

Comment:
Please get rid of the old constructor.  If people want to create a RequestContext without the software name and version information, they can easily do it explicitly with the other constructor.  Keeping this around just leads to confusion.